.github/workflows/plugin_tests.yaml

```diff
@@ -0,0 +1,46 @@
+name: Remote Plugin Tests
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  tests:
+    timeout-minutes: 45
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04, windows-latest, macos-latest]
+        pyv: ["3.10"]
+        plugin: ["dvc-s3"]
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        repository: iterative/${{ matrix.plugin }}
+        ref: main
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.pyv }}
+        cache: pip
+        cache-dependency-path: setup.cfg
+
+    - name: Install plugin + DVC@PR
+      run: |
+        pip install --upgrade pip wheel
+        pip install -e ".[tests]"
+        pip install "dvc[testing] @ git+https://github.com/iterative/dvc@${{ github.ref }}"
```

Comment:
This will not run on forks.
```yaml
- uses: actions/checkout@v3
  with:
    fetch-depth: 0
    path: dvc

- uses: actions/checkout@v3
  with:
    fetch-depth: 0
    path: ${{ matrix.plugin }}
```

Also, the order of `pip install` probably should be different, as installing `dvc[]` may install old plugin versions.
```console
pip install --upgrade pip wheel
pip install "dvc[testing]"
pip install -e "${{ matrix.plugin }}[tests]"
```

Reply:
> This will not run on forks

I'm not following here, `{{ github.ref }}` for pull requests is `refs/pull/#/merge` (not the branch ref from PR users's fork). As long as a PR is opened it doesn't matter if the PR branch is on a fork or on the main repo, github creates the `refs/pull` ref against the upstream repo (`iterative/dvc`)

> Also, the order of pip install probably should be different
👍 will fix this

Reply:
If the question was about pull requests sent to forks we could also just make it
```
dvc[testing] @ git+https://github.com/${{ github.repository }}@${{ github.ref }}
```

Reply:
Ah, i forgot that github.ref is a merge ref. I’d still suggest using checkout though, as I think it’s more explicit and simple, and does not have to go through confusing pip commands.